tv_apps/smart-home/js/home.js

```diff
@@ -322,12 +322,18 @@
     },
 
     onSearchBarShown: function() {
-      /*jshint nonew: false */
-      new MozActivity({
+      var hideSearchBar = function() {
+        document.removeEventListener('visibilitychange', hideSearchBar);
+        this.searchBar.hide();
+      }.bind(this);
+      document.addEventListener('visibilitychange', hideSearchBar);
```

Comment:
How about this one:

document.addEventListener('visibilitychange', function hideSearchBar() {
  document.removeEventListener('visibilitychange', hideSearchBar);
  this.searchBar.hide();
}.bind(this));
```